--- mail_dict.c.orig
+++ mail_dict.c
@@ -46,6 +46,7 @@
 #include <dict_mysql.h>
 #include <dict_pgsql.h>
 #include <dict_sqlite.h>
+#include <dict_redis.h>
 #include <dict_memcache.h>
 #include <mail_dict.h>
 #include <mail_params.h>
@@ -70,6 +71,9 @@
 #endif
 #ifdef HAS_SQLITE
     DICT_TYPE_SQLITE, dict_sqlite_open,
+#endif
+#ifdef HAS_REDIS
+    DICT_TYPE_REDIS, dict_redis_open,
 #endif
 #endif					/* !USE_DYNAMIC_MAPS */
     DICT_TYPE_MEMCACHE, dict_memcache_open,
